components/chart/index.tsx

```diff
@@ -56,8 +57,11 @@ function Chart({ data, token }: { config: Config; data: GraphContent[]; token?:
     const arrayOfNodes = Object.keys(currentVersion.nodes).map(
       (key) => currentVersion.nodes[key]
     );
-    const m: NodeType = arrayOfNodes.find((module) => module.id === id);
-    return m ? m : undefined;
+    const module = arrayOfNodes.find((m) => m.id === id);
+    if (module) {
+      setSearchedNode(module);
+    }
+    return module;
```

Comment:
From the name of the function "findModuleById", the responsibility of that function should only get the module by id, it should not change the state of the component

So I think the correct place to put that inside focusModule callback, so in that case, even if we passed the node in the query string, it will focus the node too which would be great.